upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -521,8 +521,9 @@ if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
     # Skopeo image is stored as "skopeo:csm-${CSM_RELEASE}"
     podman load -i "${CSM_ARTI_DIR}/vendor/skopeo.tar"
     nexus_images=$(yq r -j "${CSM_MANIFESTS_DIR}/platform.yaml" 'spec.charts.(name==cray-precache-images).values.cacheImages' | jq -r '.[] | select( . | contains("nexus"))')
+    worker_nodes=$(grep -oP "(ncn-w\w+)" /etc/hosts | sort -u)
```

Comment:
nit: match digits instead of any character.
```suggestion
    worker_nodes=$(grep -oP "(ncn-w\d+)" /etc/hosts | sort -u)
```

Reply:
It might be better to get this list of nodes from HMS instead, but this should work. The only case where it might not work, is if a system has added a node and `/etc/hosts` hasn't been updated yet (since we don't recreate it on reboot, and nothing forces updating it during hardware add).

This would get all of the XNAMES, which should be SSHable.
```bash
if IFS=$'\n' read -rd '' -a NCN_XNAMES; then
        :
      fi <<< "$(cray hsm state components list --role Management --type Node --format json | jq -r '.Components | map(.ID) | join("\n")')"
```

Reply:
This will return all nodes, not only workers, right? It's ok (although unnecessary) to have k8s masters on the list, but other nodes (e.g. storage) will return failure on `crictl` call.

Reply:
Parsing `/etc/hosts` is used in a few previous steps in this file (`UPDATE_SSH_KEYS`, `REPAIR_AND_VERIFY_CHRONY_CONFIG`), so in case if new node was not added properly to `/etc/hosts`, there will be problems anyway.